activerecord/lib/active_record/core.rb

```diff
@@ -162,9 +162,16 @@ def initialize_find_by_cache # :nodoc:
         @find_by_statement_cache = { true => Concurrent::Map.new, false => Concurrent::Map.new }
       end
 
+      def define_table_constant(child_class)
+        # We have a connection at this point, which we need for this to work
+        Table.base = child_class
+        child_class.const_set(:Table, Table)
```

Comment:
I believe this is reassigning the same `Table` module to all objects, so `*::Table` is always going to refer to the same object which will be configured to the last loaded model 😅.

What we can probably do instead is define a new a module for each model and `extend` the common behaviour.

``` suggestion
        table = Module.new
        table.extend(ActiveRecord::Table)
        table.base = child_class
        child_class.const_set(:Table, table)
```

Reply:
That would explain it! I knew I was doing something wrong here.

Reply:
Actually. We probably don't want this constant to be a module at all (one immediate problem I thought of is having columns named `name` 😬). Maybe we can make `ActiveRecord::Table` a plain `class` and here do

``` suggestion
        ActiveRecord::Table.new(self)
```

Reply:
This was my first approach actually and I didn't have much luck with it. Let me push up what I have currently and you can see how it suits you.